src/app/components/shell/router.js

```diff
@@ -109,37 +109,38 @@ function useLinkHandler() {
     return linkHandler;
 }
 
-function ImportedPage({name, fallback}) {
+function useAnalyticsPageView() {
     const history = useHistory();
     const isRedirect = history.location.state?.redirect;
-    const {fail} = useRouterContext();
-    const Component = React.useMemo(
-        () => {
-            const importFn = () => import(`~/pages/${name}/${name}`);
-
-            return React.lazy(() => retry(importFn).catch(fallback || fail));
-        },
-        [name, fallback, fail]
-    );
 
     useEffect(() => {
         window.scrollTo(0, 0);
         if (!isRedirect) {
             analytics.sendPageview();
         }
-    }, [name, isRedirect]);
-
-    return (
-        <Suspense fallback={<LoadingPlaceholder />}>
-            <Component />
-        </Suspense>
-    );
+    }, [isRedirect]);
 }
 
-function useHomeOrMyOpenStax() {
-    const user = useMyOpenStaxUser();
+// Dynamic import is persnickety about building paths; you can't just pass it
+// a string with a path, it needs to know it's building a string
+// Then webpack chunks everything that matches that string (in this case, all pages)
+// in one chunk. Not as dynamic as one might hope.
+function useImportedComponent(name) {
```

Comment:
I'm not sure if this will work, but I've also fought with this and the solution I came up with was passing a function that calls import around instead of a string.  So this could look like:
```
function useImportedComponent(importFn) {
    ...
}
```

and then be called like:
```
const page =  ImportedPage(useMemo(() => import('~/pages/foo/bar')));
```

Reply:
I had something like that when I was trying to load the General page and the Normal page in parallel. That did not go well, so I wound up back with just one import function.